app/src/main/java/io/homeassistant/companion/android/widgets/multi/MultiWidget.kt

```diff
@@ -0,0 +1,438 @@
+    }
+
+    override fun onDisabled(context: Context) {
+        // Enter relevant functionality for when the last widget is disabled
+    }
+
+    private suspend fun getWidgetRemoteViews(context: Context, appWidgetId: Int): RemoteViews {
+        // Every time AppWidgetManager.updateAppWidget(...) is called, the click listeners
+        // and label/template need to be re-assigned, or  the widget will fall back on its
+        // default layout without any click listeners being applied
+
+        val updateIntent = Intent(context, MultiWidget::class.java).apply {
+            action = UPDATE_WIDGET
+            putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId)
+        }
+        val lowerServiceIntent = Intent(context, MultiWidget::class.java).apply {
+            action = CALL_LOWER_SERVICE
+            putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId)
+        }
+        val upperServiceIntent = Intent(context, MultiWidget::class.java).apply {
+            action = CALL_UPPER_SERVICE
+            putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId)
+        }
+
+        val widget = multiWidgetDao.get(appWidgetId)
+
+        // Create an icon pack and load all drawables if a button is present
+        if (widget != null) {
+            if ((widget.upperButton == 1) || (widget.lowerButton == 1)) {
+                if (iconPack == null) {
+                    val loader = IconPackLoader(context)
+                    iconPack = createMaterialDesignIconPack(loader)
+                    iconPack!!.loadDrawables(loader.drawableLoader)
+                }
+            }
+        }
+
+        return RemoteViews(context.packageName, R.layout.widget_multi).apply {
+            // Set on-click pending intents
+            setOnClickPendingIntent(
+                R.id.widgetLabel,
+                PendingIntent.getBroadcast(
+                    context,
+                    appWidgetId,
+                    updateIntent,
+                    PendingIntent.FLAG_UPDATE_CURRENT
+                )
+            )
+            setOnClickPendingIntent(
+                R.id.widgetImageButtonUpper,
+                PendingIntent.getBroadcast(
+                    context,
+                    appWidgetId,
+                    upperServiceIntent,
+                    PendingIntent.FLAG_UPDATE_CURRENT
+                )
+            )
+            setOnClickPendingIntent(
+                R.id.widgetImageButtonLower,
+                PendingIntent.getBroadcast(
+                    context,
+                    appWidgetId,
+                    lowerServiceIntent,
+                    PendingIntent.FLAG_UPDATE_CURRENT
+                )
+            )
+
+            if (widget != null) {
+                // If there are buttons, set button icons
+                if (widget.lowerButton == 1) {
+                    val iconId = widget.lowerIconId ?: 62017 // Lightning bolt
```

Comment:
So there are two different lightning bolts, as it turns out.  The one that displays on the icon that I had originally imported as a drawable XML (the one with the flat top) seems to be 62017.  The other one, 988171, I believe is a more jagged "Harry Potter" lightning bolt.  I believe that 988171 was picked when the icon picker was changed to the mdi library.  I changed it to 62017 here so it would match the 'preview' default icon.